atlasdb-commons/src/main/java/com/palantir/util/io/AvailabilityRequirement.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.util.io;
+
+public enum AvailabilityRequirement {
+    ANY {
+        @Override
+        protected int calculateRequired(int total) {
```

Comment:
https://errorprone.info/bugpattern/ImmutableEnumChecker

Reply:
(As to why I opted for this rather than `IntFunction`)

Reply:
not actionable: This makes sense - yeah querying the predicates out of an enum is pretty awful. The example they posted is a bit of a strawman in that that isn't how I'd write this if I was told to use IntUnaryOperators or whatever, though - 

```
enum Types {
  STRING(o -> o instanceof String),
  NUMBER(o -> o instanceof Number),
  // ...

  private final Predicate<Object> hasCompatibleType; // you shouldn't be able to read me

  Types(Predicate<Object> hasCompatibleType) {
    this.hasCompatibleType = hasCompatibleType;
  }

  public final boolean hasCompatibleType(Object o) {
    // do some common logic on o
    return hasCompatibleType.test(o);
  }
}
```

is probably more natural. (What you've written is still preferable IMO though!)